client/src/ehr_components/NewPatient.tsx

```diff
@@ -0,0 +1,245 @@
+import { useHistory } from "react-router-dom";
+import { toastr } from "react-redux-toastr";
+import React, { useState } from "react";
+import Axios from "axios";
+
+import FieldRenderer from "./common_components/field-renderer";
+import { stringify } from "querystring";
+
+export interface NewPatientProps {
+  added: Function;
+}
+
+const NewPatient: React.FC<NewPatientProps> = props => {
+  const history = useHistory();
+  const [formSubmitted, setFormSubmitted] = useState(false);
+  const [formData, setFormData] = useState({
+    fullName: "",
+    dob: "",
+    phoneNumber: "",
+    address: "",
+    zipCode: "",
+    gender: "",
+    job: "",
+    maritalStatus: ""
+  });
+
+  const firebaseURl =
+    "https://idoctorpwa-default-rtdb.firebaseio.com/patients.json";
+
+  const updateFormData = (
+    fieldName: string,
+    event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
+  ) => {
+    setFormData({
+      ...formData,
+      [fieldName]: event.target.value
+    });
+  };
+
+  let fieldsMap = [
+    {

```

Comment:
FYI: as this get bigger, it make sense to move it to another file (e.g new patient fields or constant) just to keep files maintainable

Reply:
Updated! I moved the fields to a separate file, data/new-patient-fields.ts